### PR TITLE
feat: Add support for looking up changelog based on a binary package name

### PR DESCRIPTION
**Pull request by philroche**
If you do not know the source package name you can now use new `--binary-package`
and `--binary-package-architecture` args to find the changelog for the source package
that builds the specified binary package name.

### CODE REVIEW

File: ubuntu_package_changelog/__init__.py [hunk 1]
```diff
@@ -19,9 +19,30 @@ def _lp_get_changelog_url(args, lp):
...
+            exact_match=True,
+            binary_name=args.package,
+            pocket=pocket,
+            distro_arch_series=lp_arch_series,
+            status="Published",
+            order_by_date=True,
+        )
+        if len(binaries):
+            for binary in binaries:
+                print(binary.source_package_version)
```
Comment (toabctl) [inline ubuntu_package_changelog/__init__.py]: can you drop those print? only for debugging, right?

File: ubuntu_package_changelog/__init__.py [hunk 2]
```diff
@@ -48,6 +69,16 @@ def _parser():
                         type=_args_validate_ppa_name)
     parser.add_argument('--entries', help='number of changelog entries to get from the '
                         'changelog. 0 mean all. Default: %(default)s', type=int, default=1)
+    parser.add_argument('--binary-package',
```
Comment (toabctl) [inline ubuntu_package_changelog/__init__.py]: hm. those new flags do break the current CLI. the CLI currently expects a `package` name as the last parameter. That parameter is useless if you use now `--binary-package`. 
What about adding an automatic fallback? So if the given `package` is not a source package, try to find the binary package and get the source package from that?

**Review by toabctl — CHANGES_REQUESTED** [review]

File: ubuntu_package_changelog/__init__.py (hunk 2 above)
Comment (philroche) [reply ubuntu_package_changelog/__init__.py]: package is still used when --binary-package is passed. --binary-package is a flag and is used to do decide whether to do a lookup with package to find the source package before trying to get the changelog

Comment (toabctl) [reply ubuntu_package_changelog/__init__.py:72]: ah, so I missunderstood it. thanks!

File: ubuntu_package_changelog/__init__.py [hunk 3]
```diff
@@ -48,6 +69,16 @@ def _parser():
...
+                        help='Set this option if you wish to get the changelog for the '
+                        'source package of the specified binary package name. '
+                        'Default: %(default)s',
+                        action='store_true', default=False)
+    parser.add_argument('--binary-package-architecture',
```
Comment (toabctl) [inline ubuntu_package_changelog/__init__.py]: would it be possible to drop that flag and just use whatever architecture a binary package has? The source package and changelog is not different between the architectures.

**Comment by philroche** [comment]
--binary-package-architecture and --binary-package now removed and binary lookup is only done if published sources are not found.

**Pull request merged**